### DISCUSSION TITLE
RFC: Simplify the returns system

### DISCUSSION THREAD

**Discussion by seand7565** [solidusio/solidus#4596] [New Features or Ideas]
Right now on vanilla Solidus, it takes ~17 clicks to do a simple return. The system is pretty convoluted and I don't think a lot of companies need that much complication. We did a brief poll at SolidusConf, and only a very small amount of people still use the default returns system - most people have overridden it with a simpler solution.

Also: The way the returns system is written right now makes it very difficult to solve the inventory units issue ( #2342 ). Replacing it with something simpler would make that issue easier to solve.

i'd love to hear some thoughts on a simpler returns system.

**Comment by AdnanTheExcellent** [comment]
I could possibly work on this. We have #2342 solved in our code and we've simplified the return process so its not as convoluted. I could provide some screenshots to see if people agree with our changes before i start working on extracting the code into a PR.

**Comment by seand7565** [comment]
Sure, we'd love to see what you've come up with. Initially, I think it makes sense to rip out the existing returns system into an extension - Solidus Legacy Returns or something - and replace with a simple "return_item" object that has quantity, state, and possibly price (if return price is different than selling price)... Then you'd just need a page on the order to manage them. If there was a way to migrate the old return system into the new one, that would be a huge plus as well.

**Comment by jarednorman** [comment]
Call it `solidus-complicated-bonobos-returns`! 😂
